crates/swift-bridge-ir/src/bridged_type/shared_struct.rs

```diff
@@ -19,18 +21,37 @@ pub(crate) struct SharedStruct {
     pub fields: StructFields,
     pub swift_name: Option<LitStr>,
     pub already_declared: bool,
+    pub is_tuple: bool,
```

Comment:
We don't want to mix unrelated states.

For example, it doesn't make sense that a type that is being used for a tuple would have a `StructSwiftRepr`.

This will just lead to confusion and code that has weird if statements where we sometimes care about some fields and sometimes don't.

---

A tuple is really just a `StructFields::Unnamed`. All of these other fields don't really apply to tuples.

I think that we should have 

```rust
// bridged_type/built_in_tuple.rs
pub(crate) struct BuiltInTuple(UnnamedStructFields);

impl BridgeableType for BuiltInTuple { /* .. */ }
```

And then change the existing `StructFields` enum to

```rust
#[derive(Debug, PartialEq, Clone)]
pub(crate) enum StructFields {
    Named(Vec<NamedStructField>),
    // Notice that this is now `UnnamedStructFields`.
    Unnamed(UnnamedStructFields),
    Unit,
}

// Notice this new type.
pub(crate) struct UnnamedStructFields(Vec<UnnamedStructField>);
```

Then any re-usable code that we need between tuples and tuple structs can be put in methods on the `UnnamedStructFields` type.